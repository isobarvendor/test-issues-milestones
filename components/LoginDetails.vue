<template>
<div class="login-details">
  <div class="title-image"><img src="img/icons/coke.png" />
      <p>Sign up with us and gain access to our rewards catalogue where you can redeem prizes.</p>
  </div>
  <div class="continue" v-if="$mq == 'xl'">
      <p>Continue with</p>
      <div class="icons">
          <div v-on:click="redirectTo('line')" v-if="social.line"><span><img src="img/icons/line.png" /></span></div>
          <div v-on:click="redirectTo('facebook')" v-if="social.facebook" ><span><img src="img/icons/fb.png" /></span></div>
          <div v-on:click="redirectTo('google')"  v-if="social.google"><span><img src="img/icons/google.png" /></span></div>
          <div v-on:click="redirectTo('email')"  v-if="social.email"><span><img src="img/icons/email.png" /></span></div>
      </div>
      <p> or </p>
  </div>
  <div class="login-box column center" v-if="$mq == 'sm' || $mq == 'md' || $mq == 'lg'">
    <a class="social-button fb" v-on:click="redirectTo('facebook')"><i class="ic ic-fb" /><span>Continue with Facebook</span></a>
    <a class="social-button line"  v-on:click="redirectTo('line')"  ><i class="ic ic-line" /><span>Continue with LINE</span></a>
    <a class="social-button google"  v-on:click="redirectTo('google')"><i class="ic ic-google" /><span>Continue with Google</span></a>
    <a class="social-button email"  v-on:click="redirectTo('email')"><i class="ic ic-email" /><span>Continue with email</span></a>
  </div>
  <transition name="fade">
    <div class="details" v-if="show">
        <div class="input-container">
            <v-text-field label="Email address" v-validate="'required|email'" name="email"  v-model="login.email"></v-text-field>
            <span class="error-message">{{ errors.first('email') }}</span>
            <v-text-field label="Password"  v-validate="'required'" name="password"   :type="'password'" v-model="login.password"></v-text-field>
            <span class="error-message">{{ errors.first('password') }}</span>
        </div>
        <div class="button-container">
           <div class="error-message" v-if="errorMessage" v-html="errorMessage"></div>
            <a  class="button" v-on:click="userLogin()" >Login</a>
            <p>Forget Password</p>
            <p  v-on:click="userSignUp()" >Sign Up</p>
        </div>

    </div>
  </transition>
</div>
</template>

<script>
import { LOGIN } from '@/store/action_types';
export default {
  name: "LoginDetails",
  inject: ['$validator'],
  components: {
  },
  data() {
    return {
        show:true,
        login: {
          email:'',
          password:''
        },
        errorMessage:null
    };
  },
  props: {
    data: null,
    social: null
  },
  computed:{
  },
  methods:{
    testing(){
        return this.show = true;
    },
    redirectTo(source){
      location.href="http://localhost:8080/api/oauth2/authorize/"+source+"?redirect_uri=http://localhost:3000/settoken";
    },
    async userLogin() {
       this.$validator.validateAll().then((valid) => {
         if(valid){
             this.errorMessage=null;
            this.$store.dispatch(LOGIN,this.login)
            .then(function (response) {
              console.log(response);
              this.$router.push('/account');
            })
            .catch((error) =>{
              if(error.response.data.status=='401'){
                this.errorMessage='Please enter the correct email/password';
              }
            })
         }

       });

    },
     userSignUp() {
         this.$router.push('/register');

    }
  },
  beforeMount() {},
  mounted(){
      if(this.$mq == 'sm' || this.$mq == 'md' || this.$mq == 'lg'){
        this.show=false;
       }
  }
};
</script>

<style lang="scss">
.login-details{
    display: flex;
    flex-direction: column;
    align-items: center;
    .error-message{
      color:red;
    }
    .title-image{
        img{
            width: 230px;
            margin: auto;
            display: table;
            margin-bottom: 25px;
        }
        width: 50%;
        margin-bottom: 20px;
        text-align: center;
        line-height: 24px;
    }
    .continue{
        text-align: center;
        p{
            margin-bottom: 15px;
            margin-top: 15px;
        }
        .icons{
            width: 350px;
            display: flex;
            justify-content: space-evenly;
        }
    }
    .details{
        width: 50%;
    }
    .v-text-field__slot{
        border-bottom: 1px solid #585858;
    }
    .theme--light.v-label{
        color: #585858;
    }
    .theme--light.v-input input{
        color: #ffffff;
    }
    .button-container{
            margin-top: 40px;
            text-align: center;
            a.button{
                width: 230px;
                display: block;
                margin: auto;
            }
            p{
                margin-top: 40px;
            }
        }

    @media only screen and (max-width: 1199px) {
        .title-image{
            width: 100%;
        }
        .login-box{
            margin-top:25px;
            width:100%;
        }
        .details{
            width: 80%;
            margin-top: 40px;
        }
        .fade-enter-active, .fade-leave-active {
            transition: opacity .5s;
        }
        .fade-enter, .fade-leave-to /* .fade-leave-active below version 2.1.8 */ {
            opacity: 0;
        }
    }
}
</style>
