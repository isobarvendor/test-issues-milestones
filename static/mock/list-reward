[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
