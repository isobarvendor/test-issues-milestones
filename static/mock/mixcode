{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
