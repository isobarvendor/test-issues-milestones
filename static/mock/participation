{"id":16,"questionId":1,"answerId":3,"participationId":17}
