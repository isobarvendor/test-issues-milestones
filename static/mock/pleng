{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}