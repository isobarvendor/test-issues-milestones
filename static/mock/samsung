{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>ទូរស័ព្ទសាមសុង Galaxy Note20 Ultra</p>",
      "priority": 1,
      "prizeId": "ks8ns5hv",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8ns5hv/1.CAM prize-N20 Ultra@2x.png",
      "totalAmount": 20,
      "minAge": 13,
      "name": "Samsung Galaxy Note20 Ultra",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះទូរស័ព្ទសាមសុង Galaxy Note20 Ultra! សូមទាក់ទងទៅលេខ 1-800-20-2222 (មិនគិតថ្លៃសេវាទូរស័ព្ទ ២៤/៧) ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}