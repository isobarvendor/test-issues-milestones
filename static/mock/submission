{
            "burnResult": [
                {
                    "pincode": "0J6TC2VP5",
                    "programId": "453269",
                    "lotId": "445580",
                    "burned": true
                }
            ],
            "participationId": "1ylazdy9kkxmpa53",
            "instantWinResult": {
                "winner": true,
              "redeemedPrize": {
                    "prizeId": "kkqhy9so",
                    "voucherCode": "Voucher891",
                    "status": "claimed",
                    "expiryDate": 1620018590000,
                    "name": "JOOX VIP PASS",
                    "description":"Conglratulations",
                    "shortDescription": "http://google.com",
                    "redeemDescription": "http://abcd",
                    "imgUrl": "/img/landing/week 1 prize.png",
                    "barcodeType": 1,
                    "emailSent": false,
                    "emailMessage": "Email could not be sent"
                }
            }
        }
