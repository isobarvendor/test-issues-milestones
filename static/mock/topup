{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>លេខកូដកាតសម្រាប់បញ្ចូលទឹកប្រាក់ 5 ដុល្លា ដែលមានរយៈពេល 30 ថ្ងៃ</p>",
      "priority": 1,
      "prizeId": "ks8nusf9",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8nusf9/3.CAM prizes-Top up$5@2x.png",
      "totalAmount": 500,
      "minAge": 13,
      "name": "Top up card/code $5",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះកោសកាតប្រព័ន្ធទូរស័ព្ទ Smart 5 ដុល្លា! នេះជាលេខកូដសម្រាប់បញ្ចូលលុយទូរស័ព្ទ សូមចុចលេខ *888*លេខកូដ ១៣ខ្ទង់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}