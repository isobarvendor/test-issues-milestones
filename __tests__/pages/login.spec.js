import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';

import Login from '@/pages/login.vue';

import LoginDetails from '@/components/LoginDetails'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";



const $route = {
  name: "login"
}

let localVue, store;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getLoginAccount(){
          return {
            name:"ron",
            email:"ron@dsw.com",
            imageUrl:"tes"
          }
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      }
    });
  });

  const factory = (route) => {
    return mount(Login, {
      mocks: {
        $route : route
      },
      localVue, store,
    });
  };



describe('login', () => {


  test('renders a LoginDetails', () => {
    const wrapper = factory($route);
    const component= wrapper.findComponent(LoginDetails);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory($route);
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory($route);
    expect(wrapper.html()).toMatchSnapshot();
  });

});
