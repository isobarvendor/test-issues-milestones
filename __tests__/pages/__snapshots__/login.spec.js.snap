// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`login renders properly 1`] = `
<div id="main" class="login">
  <div class="close"><img src="/img/icons/close.png"></div>
  <div class="wrapper">
    <div class="container">
      <div class="login-details">
        <div class="title-image"><img src="img/icons/coke.png">
          <p>
            Sign up with us and gain access to our rewards catalogue where you can redeem prizes.
          </p>
        </div>
        <!---->
        <!---->
        <transition-stub name="fade">
          <div class="details">
            <div class="input-container">
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-31" class="v-label theme--light" style="right: 0px; position: absolute;">Email address</label><input name="email" id="input-31" type="text"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
              <div class="v-input theme--light v-text-field">
                <div class="v-input__control">
                  <div class="v-input__slot">
                    <div class="v-text-field__slot"><label for="input-34" class="v-label theme--light" style="right: 0px; position: absolute;">Password</label><input name="password" id="input-34" type="password"></div>
                  </div>
                  <div class="v-text-field__details">
                    <div class="v-messages theme--light">
                      <transition-group-stub tag="div" name="message-transition" class="v-messages__wrapper"></transition-group-stub>
                    </div>
                  </div>
                </div>
              </div> <span class="error-message"></span>
            </div>
            <div class="button-container">
              <!---->
              <div class="btn-area"><a class="button">Login</a></div>
              <p>Sign Up</p>
            </div>
          </div>
        </transition-stub>
      </div>
    </div>
  </div>
</div>
`;
